crates/amaru-sim/Cargo.toml

```
@@ -0,0 +1,49 @@
+[package]
+name = "amaru-sim"
+version.workspace = true
+edition.workspace = true
+description.workspace = true
+license.workspace = true
+authors.workspace = true
+repository.workspace = true
+homepage.workspace = true
+documentation.workspace = true
+rust-version.workspace = true
+
+[package.metadata.cargo-machete]
+ignored = ["maelstrom-node"]
+
+[dependencies]
+async-trait.workspace = true
+clap = { workspace = true, features = ["derive", "env"] }
+gasket = { workspace = true, features = ["derive"] }
+hex.workspace = true
+miette = { workspace = true, features = ["fancy"] }
+pallas-codec.workspace = true
+pallas-crypto.workspace = true
+pallas-network.workspace = true
+pallas-primitives.workspace = true
+tokio = { workspace = true, features = ["rt", "rt-multi-thread", "signal"] }
+tokio-util.workspace = true
+tracing-subscriber = { workspace = true, features = ["env-filter", "std", "json"] }
+tracing.workspace = true
+serde.workspace = true
+serde_json.workspace = true
+maelstrom-node = "0.1.6"
```

```suggestion
maelstrom-node.workspace = true
```

Implying that the version is also moved upstream to the `Cargo.toml`

---

I am not sure yet this is needed or worthwhile: What this crate provides for us is not much really, hence why I am keeping it there for now

---

This is just about keeping things tidy. We keep all versions in one place, in the top-level Cargo.toml. Otherwise, it'll rapidly become a mess mainly because it then becomes unclear whether a dependency shall be specified for a crate only, or for the workspace. What's the rule? It kind of becomes an arbitrary decision and that's certain to become inconsistent and unclear from one person to another. So, we go by one simple rule: all dependencies versions are specified once for the entire workspace. 

Note that doesn't mean the dependency gets pulled for ALL crates; not at all. It's just a namespace where they live, and they only get pulled if effectively required.

---

Fair enough, I will pull it up